### PR TITLE
Feat account service backend integration

### PR DESCRIPTION
**Pull request by mattheworris**
We would like to wire up the account-service into the social-app-template.

## Notes on Changes:
- There were some changes on the front-end due to some migration of `dsnpId` to `msaId` needed to overcome type errors. There are still some references to `dsnpId` that need to be migrated and will be handled in a follow-up issue.

## AC:

- [x] I should be able to use the account service to sign in to the social-app-template
- [x] I should be able to use the account service to sign up (_including handle and permissions_) to the social-app-template
- [x] I should be routed to the index/home.landing page after I have signed in/up
- [x] I should see my handle in the header (_like my username on normal web2 applications_) 
- [x] I should be able to disconnect (or log out) from the social app template and be routed back to the sign in/up page

## Notes:

- This assumes siwf is being used
- This also assumes a provider has already been configured via the provider-dashboard and done via env vars 

Repos:
https://github.com/AmplicaLabs/account-service
https://github.com/AmplicaLabs/siwf
https://github.com/frequency-chain/provider-dashboard

### CODE REVIEW

File: backend/controllers/AuthController.ts
Comment (mattheworris) [inline backend/controllers/AuthController.ts:18]: The current flow of the front-end requires that back-end has the public key to create the authToken. 
However, during the sign-up process, the front-end has to wait on the back-end to send a transaction to the blockchain to create the msaId, so the authToken is not available yet.

File: backend/environment/env.social-app-backend.template
Comment (mattheworris) [inline backend/environment/env.social-app-backend.template:8]: It seems like this initial value is needed to pass CI

File: backend/services/AccountService.ts
Comment (mattheworris) [inline backend/services/AccountService.ts:85]: Follow up: creating an issue to update msaId to be the same type everywhere, currently some are string and some are number.

File: backend/controllers/AuthController.ts
Comment (JoeCap08055) [inline backend/controllers/AuthController.ts:41]: At some point I'd like to review what exactly is really needed in this response. Okay for now.

File: backend/environment/env.social-app-backend.template
Comment (JoeCap08055) [reply backend/environment/env.social-app-backend.template:8]: Fine for now. Actually the way I solved it was to comment out the env var. It's allowed to be absent, just not empty. I'll fix it in an upcoming PR.

File: backend/services/AccountService.ts
Comment (JoeCap08055) [reply backend/services/AccountService.ts:87]: 👍🏻 
correct type should be 'string' (or bigint, but we use string most other places...)

File: backend/services/AccountService.ts
Comment (JoeCap08055) [inline backend/services/AccountService.ts:107]: I think maybe move `referenceIdsReceived` out of the controller class and into a service; seems like it would be more consistent with the design. `WebhookController` is meant to be super-lightweight, as it's serving webhooks across all domains.

File: backend/services/AccountService.ts
Comment (JoeCap08055) [inline backend/services/AccountService.ts:139]: we're in the `AccountService` here already, so we can ditch `await AccountService.getInstance()` and replace with `this`

File: backend/services/AccountService.ts
Comment (JoeCap08055) [inline backend/services/AccountService.ts:155]: nit: this has caused confusion in the past--maybe rename to `chainApi`, to distinguish from a REST api?

File: backend/services/TokenAuth.ts
Comment (JoeCap08055) [inline backend/services/TokenAuth.ts:20]: 🔥 thanks for all the JSDoc comments

File: frontend/src/dsnpLink.ts
Comment (JoeCap08055) [inline frontend/src/dsnpLink.ts:204]: yes... is this a generated file?

Comment (mattheworris) [reply frontend/src/dsnpLink.ts:204]: Yes, `dsnpLink.ts` is generated from `openapi.json`, see frontend `npm run gen:client`

**Review by saraswatpuneet — COMMENTED** [review]
Having worked with PR there are few comments

* We can comment out broadcast tests and open an issue to modify TokenAuth to be mockable. Basically mocking a function which is called by another function within file is a pitfall of vitest , the function in question is `getAccountAuth` 

* Tested locally have no issues

Running some more tests before I approve it

**Review by saraswatpuneet — APPROVED** [review]
Lgtm! Ship it when ready

File: backend/controllers/AuthController.ts
Comment (mattheworris) [reply backend/controllers/AuthController.ts:44]: #43

File: frontend/src/dsnpLink.ts
Comment (JoeCap08055) [reply frontend/src/dsnpLink.ts:204]: See, this is the problem with generated code. We'll have to revisit the whole question of tooling around this app template. The backend is no longer generated from `openapi.json`, so we may encounter drift. This is probably a post-1.0 thing, but this setup is very brittle & should be addressed.

File: backend/services/AccountWebhookService.ts
Comment (JoeCap08055) [inline backend/services/AccountWebhookService.ts:16]: Let's create an issue to put the webhook request & response payload types into an OpenAPI spec document in the `account-service` repo, copy to this repo & generate types from it. (I can show you what I did in `content-watcher` for the same)

File: backend/services/AccountWebhookService.ts
Comment (JoeCap08055) [inline backend/services/AccountWebhookService.ts:20]: nit: create a type for the values of this Map

File: backend/services/AccountWebhookService.ts
Comment (mattheworris) [reply backend/services/AccountWebhookService.ts:16]: Cool, there's already an openApi spec in `account-service/docs/webhook.endpoints.json` 
#44

**Pull request merged**
